File: project-euler/problem-3/main.py
""" Project Euler Problem 3 - Largest Prime Factor.

The prime factors of 13195 are 5, 7, 13 and 29.

What is te largest prime factor of 600851475143?
"""


def largestPrimeFactor(target: int = 600851475143) -> int:
    num = target
    largestFactor = 0
    counter = 2

    while counter ** 2 <= num:
        if (num % counter) == 0:
            num /= counter
            continue

        counter += 1

    if num > largestFactor:
        largestFactor = num

    return int(largestFactor)


if __name__ == "__main__":
    print(f"{largestPrimeFactor() = }")

File: project-euler/problem-3/main.js
"use strict";
/**
 * Project Euler Problem 3 - Largest Prime Factor.
 *
 * The prime factors of 13195 are 5, 7, 13 and 29.
 *
 * What is the largest prime factor of 600851475143?
 */
function largestPrimeFactor(target = 600851475143) {
    let num = target;
    let largestFactor = 0;
    let counter = 2;
    while (counter ** 2 <= num) {
        if (num % counter === 0) {
            num = num / counter;
            continue;
        }
        counter++;
    }
    if (num > largestFactor) {
        largestFactor = num;
    }
    return largestFactor;
}
console.log(`largestPrimeFactor() = ${largestPrimeFactor()}`);
